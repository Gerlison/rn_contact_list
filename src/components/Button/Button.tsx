import React, { memo } from 'react';
import { TouchableOpacityProps } from 'react-native';
import styled from 'styled-components/native';
import Text from '../Text';

interface Props extends TouchableOpacityProps {
  label: string;
}

const Button = ({ label, ...props }: Props): JSX.Element => {
  return (
    <S.Touchable {...props}>
      <Text variant="Poppins-SemiBold" size={18} color="#fff">
        {label.toUpperCase()}
      </Text>
    </S.Touchable>
  );
};

const S = {
  Touchable: styled.TouchableOpacity`
    width: 100%;
    align-items: center;
    padding: 18px;
    margin-top: 32px;
    border-radius: 20px;
    background-color: #7c37a6;
  `,
};

export default memo(Button);
