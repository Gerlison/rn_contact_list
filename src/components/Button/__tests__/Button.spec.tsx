import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import Button from '../Button';

describe('Button', () => {
  it('SHOULD match snapshot', () => {
    const component = render(<Button label="foo" />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('SHOULD call onPress on click', () => {
    const mockFn = jest.fn();
    const component = render(<Button label="foo" onPress={mockFn} />);
    fireEvent.press(component.getByText('FOO'));
    expect(mockFn).toBeCalled();
  });
});
