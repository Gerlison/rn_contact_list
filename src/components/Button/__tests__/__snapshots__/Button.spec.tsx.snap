// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button SHOULD match snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#7c37a6",
      "borderRadius": 20,
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 18,
      "paddingLeft": 18,
      "paddingRight": 18,
      "paddingTop": 18,
      "width": "100%",
    }
  }
>
  <Text
    color="#fff"
    family="Poppins-SemiBold"
    size={18}
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "Poppins-SemiBold",
          "fontSize": 18,
        },
      ]
    }
  >
    FOO
  </Text>
</View>
`;
